cfn-resources/organization/cmd/resource/resource.go

```diff
@@ -0,0 +1,365 @@
+	setup()
+
+	modelValidation := validator.ValidateModel(CreateRequiredFields, currentModel)
+	if modelValidation != nil {
+		return *modelValidation, nil
+	}
+
+	// Create atlas client
+	if currentModel.Profile == nil || *currentModel.Profile == "" {
+		currentModel.Profile = util.Pointer(profile.DefaultProfile)
+	}
+
+	atlas, peErr := util.NewAtlasClient(&req, currentModel.Profile)
+	if peErr != nil {
+		return *peErr, nil
+	}
+
+	_, _, err := secrets.Get(&req, *currentModel.AwsSecretName)
+	if err != nil {
+		// Delete the APIKey from Atlas
+		_, _ = logger.Warnf("error : no Secret exists with %s", *currentModel.AwsSecretName)
+		response := &http.Response{StatusCode: http.StatusBadRequest}
+		return handleError(response, constants.CREATE, err)
+	}
+
+	apikeyInputs := setAPIkeyInputs(currentModel)
+
+	// Set the roles from model
+	orgInput := &atlasSDK.CreateOrganizationRequest{
+		ApiKey:     apikeyInputs,
+		OrgOwnerId: currentModel.OrgOwnerId,
+		Name:       *currentModel.Name,
+	}
+	if currentModel.FederatedSettingsId != nil {
+		orgInput.FederationSettingsId = currentModel.FederatedSettingsId
+	}
+	orgCreateRequest := atlas.AtlasV2.OrganizationsApi.CreateOrganization(
+		context.Background(),
+		orgInput,
+	)
+	org, response, err := orgCreateRequest.Execute()
+
+	defer closeResponse(response)
+	if err != nil {
+		return handleError(response, constants.CREATE, err)
+	}
+
+	// Read response
+	currentModel.OrgId = org.Organization.Id
+
+	// Save PrivateKey in AWS SecretManager
+	secret := OrgProfile{OrgID: *currentModel.OrgId, PublicKey: *org.ApiKey.PublicKey, PrivateKey: *org.ApiKey.PrivateKey, BaseURL: atlas.Config.BaseURL}
+	_, _, err = secrets.PutSecret(&req, *currentModel.AwsSecretName, secret, currentModel.APIKey.Description)
+	if err != nil {
+		// Delete the APIKey from Atlas
+		response = &http.Response{StatusCode: http.StatusInternalServerError}
+		return handleError(response, constants.CREATE, err)
+	}
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "Create Completed",
+		ResourceModel:   currentModel}, nil
+}
+
+// Read handles the Read event from the Cloudformation service.
+func Read(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+
+	modelValidation := validator.ValidateModel(ReadRequiredFields, currentModel)
+	if modelValidation != nil {
+		return *modelValidation, nil
+	}
+
+	atlas, peErr := util.NewAtlasV2OnlyClient(&req, currentModel.AwsSecretName, false)
+	if peErr != nil {
+		return *peErr, nil
+	}
+
+	apiKeyUserDetails, response, err := currentModel.getOrgDetails(atlas, currentModel)
+
+	defer closeResponse(response)
+	if err != nil {
+		if response.StatusCode == http.StatusUnauthorized {
+			return handler.ProgressEvent{
+				OperationStatus:  handler.Failed,
+				Message:          "Not found",
+				HandlerErrorCode: cloudformation.HandlerErrorCodeNotFound}, nil
+		}
```

Comment:
unless I am missing something this is not needed because the same logic is in the same `handleError` method.

Reply:
@marcosuma  If the org is not available the API is responding with Unauthorized error. So CFN TEST will fail if it responds with unauthorized instead of NOT FOUND(404).

Reply:
What I am saying is that if you go to the `handleError` method you'll find this if statement:

```
	if response.StatusCode == http.StatusUnauthorized {
		return handler.ProgressEvent{
			OperationStatus:  handler.Failed,
			Message:          "Not found",
			HandlerErrorCode: cloudformation.HandlerErrorCodeNotFound}, nil
	}
```

which is exactly the same you wrote in this code change..

Reply:
@marcosuma : that is the duplicate code, I just updated